### PR TITLE
Fix periodic

### PR DESCRIPTION
**Pull request by MartinKarp**
Seems like there was an issue that the facets belonging to a periodic facet could be changed by a neighboring element, but then not added to the table correctly when generating connectivity. 

This PR changes this. For future developments, we should note though that if the entire side is not periodic but a mesh would change from periodic to some other BC along a boundary, this is not necessarily well defined if the element it changes on is not guaranteed to be on the same rank.

### CODE REVIEW

**Comment by MartinKarp** [comment]
Victor, please try out this PR for your turbine mesh

**Comment by MartinKarp** [comment]
I dunno what we want as intentional behavior here, but as things stands it is not valid to create a field and dofmap without doing generate_conn at least in the mesh

**Comment by MartinKarp** [comment]
I think the issue of connecting the topological and geometric data really shows now in the tests. A nicer solution might be that if one inits a of map, it checks the mesh and requires that the mesh has created its connectivity data. Right now one gets obscure issues instead as the dofmap assumes information about facets, edges and how they are connected.

**Comment by timofeymukha** [comment]
I've never looked into this stuff, so will be difficult to review. @adampep , could you take a look?

File: src/mesh/mesh.f90 [hunk 1]
```diff
@@ -1902,6 +1876,7 @@ function mesh_get_local_facet(this, f) result(local_id)
     integer :: local_id
 
     if (this%htf%get(f, local_id) .gt. 0) then
+       print *, 'error', f
        call neko_error('Invalid global id (local facet)')
```
Comment (njansson) [inline src/mesh/mesh.f90:1880]:
```suggestion
       call neko_error('Invalid global id (local facet) ' // f)
```

File: src/mesh/mesh.f90 [hunk 2]
```diff
@@ -381,8 +381,11 @@ end subroutine mesh_free
   subroutine mesh_finalize(this)
     class(mesh_t), target, intent(inout) :: this
     integer :: i
-
-
+    class(element_t), pointer :: ep
+    type(tuple_i4_t) :: e
+    type(tuple4_i4_t) :: f
+    integer :: p_local_idx, res
+
```
Comment (njansson) [inline src/mesh/mesh.f90:388]: are these some left overs? Doesn't seems to be used

**Comment by vbaconnet** [comment]
I am running it now and it seems to work, I will approve this once I have poked around a bit more :)

File: src/mesh/mesh.f90 (hunk 2 above)
Comment (MartinKarp) [reply src/mesh/mesh.f90:388]: Will check

File: src/mesh/mesh.f90 [hunk 3]
```diff
@@ -558,6 +605,7 @@ subroutine mesh_generate_conn(this)
     !
     ! Find all external (between PEs) boundaries
     !
+    print *, pe_size
```
Comment (vbaconnet) [inline src/mesh/mesh.f90:608]:
```suggestion
```

**Review by vbaconnet — APPROVED** [review]

**Review by njansson — APPROVED** [review]

**Pull request merged**